openmetadata-ui/src/main/resources/ui/src/components/common/CustomPropertyTable/CustomPropertyTable.tsx

```diff
@@ -238,15 +244,35 @@ export const CustomPropertyTable = <T extends ExtentionEntitiesKeys>({
     !isUndefined(entityDetails?.extension) ? (
     <ExtensionTable extension={entityDetails?.extension} />
   ) : (
-    <Table
-      bordered
-      columns={tableColumn}
-      data-testid="custom-properties-table"
-      dataSource={entityTypeDetail.customProperties?.slice(0, maxDataCap)}
-      loading={entityTypeDetailLoading}
-      pagination={false}
-      rowKey="name"
-      size="small"
-    />
+    <>
+      <div className="d-flex justify-between m-b-xs">
+        <Typography.Text className="right-panel-label">
+          {t('label.custom-property-plural')}
+        </Typography.Text>
+        {entityTypeDetail.customProperties &&
+          maxDataCap &&
+          entityTypeDetail.customProperties?.length >= maxDataCap &&
+          entityDetails.fullyQualifiedName && (
+            <Link
+              to={getEntityDetailLink(
+                entityType,
+                entityDetails.fullyQualifiedName,
+                EntityTabs.CUSTOM_PROPERTIES
+              )}>
+              {t('label.view-all')}
+            </Link>
+          )}
+      </div>
+      <Table
+        bordered
+        columns={tableColumn}
+        data-testid="custom-properties-table"
+        dataSource={entityTypeDetail.customProperties?.slice(0, maxDataCap)}
```

Comment:
entityType can be `empty` object so we should have an optional check for `customProperties`.

```suggestion
        dataSource={entityTypeDetail?.customProperties?.slice(0, maxDataCap)}
```